### ISSUE TITLE
[request] add raw events to the emitted events

### ISSUE THREAD

**Issue by benb7760** [swimlane/ngx-charts#648]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**

When using activate/deactivate/select events, there is no access to the raw event (or the element that triggered the event)

**Expected behavior**

Include the raw event (or trigger element) when the events are fired

**What is the motivation / use case for changing the behavior?**

In our environment we have a common popover component which we use to display detail. We would like to use the activate/deactivate events for our data to trigger our popover as a custom hover. At the moment, because we have no access to the element which triggered the event, we cannot choose an accurate position from which to originate the popover

**Comment by benb7760** [comment]
@marjan-georgiev I may have some time to look at this myself in the next few weeks. Do you have any preference as to the implementation? Two options I'm thinking of right now would be;

1. Simply adjust the emitters to fire `{rawEvent: rawEvent, data: this.data}` instead of `this.data`
2. Add proper classes to represent the events - with property 'rawEvent' and property 'data' (in some cases I see we also emit 'activeEntries', I could include this also)

Are you happy with either of these, or would you prefer an alternate approach?

**Comment by marjan-georgiev** [comment]
@benb7760 yes, I would just add the `rawEvent` as property on the object we are emitting. I would keep the `series` and `value` properties on the main object so we do not break existing functionality. 

We are also going to be emitting an extra `data` object, which would contain additional data that was passed to the chart, but not used for the visualization itself (like IDs). This will be part of another PR though.

**Comment by benb7760** [comment]
@marjan-georgiev I'm working on this now - I notice that at present, click events for legends - they don't output an object, just the string name. I can't include the rawEvent here without making a breaking change - would your preference be that I skip this entirely, possibly including it in a future PR?

**Comment by marjan-georgiev** [comment]
@benb7760 great! 

I'm fine changing that and replacing the string with an object, but please make sure to list it as a breaking change in the PR, so we can release a major version.
